x-pack/solutions/security/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -365,7 +421,58 @@ export const AlertsTableComponent: FC<DetectionEngineAlertTableProps> = ({
       {graphOverlay}
       <FullWidthFlexGroupTable $visible={!graphEventId && graphOverlay == null} gutterSize="none">
         <StatefulEventContext.Provider value={activeStatefulEventContext}>
-          <EuiDataGridContainer hideLastPage={false}>{AlertTable}</EuiDataGridContainer>
+          <EuiDataGridContainer hideLastPage={false}>
+            <AlertsTable<SecurityAlertsTableContext>
+              ref={alertsTableRef}
+              // Stores separate configuration based on the view of the table
+              id={id ?? `detection-engine-alert-table-${tableType}-${tableView}`}
+              ruleTypeIds={SECURITY_SOLUTION_RULE_TYPE_IDS}
+              consumers={ALERT_TABLE_CONSUMERS}
+              query={finalBoolQuery}
+              initialSort={initialSort}
+              casesConfiguration={casesConfiguration}
+              gridStyle={gridStyle}
+              shouldHighlightRow={shouldHighlightRow}
+              rowHeightsOptions={rowHeightsOptions}
+              columns={finalColumns}
+              browserFields={finalBrowserFields}
+              onUpdate={onUpdate}
+              additionalContext={additionalContext}
+              // if records are too less, we don't want table to be of fixed height.
+              // it should shrink to the content height.
+              // Height setting enables/disables virtualization depending on fixed/undefined height values respectively.
+              height={count >= 20 ? `${DEFAULT_DATA_GRID_HEIGHT}px` : undefined}
+              initialPageSize={50}
+              runtimeMappings={sourcererDataView?.runtimeFieldMap as RunTimeMappings}
+              toolbarVisibility={toolbarVisibility}
+              dynamicRowHeight={isEventRenderedView}
+              renderCellValue={CellValue}
+              renderActionsCell={ActionsCell}
+              renderAdditionalToolbarControls={
+                tableType !== TableId.alertsOnCasePage ? AdditionalToolbarControls : undefined
+              }
+              actionsColumnWidth={leadingControlColumn.width}
+              getBulkActions={getBulkActions}
+              fieldsBrowserOptions={
+                tableType === TableId.alertsOnAlertsPage ||
+                tableType === TableId.alertsOnRuleDetailsPage
+                  ? fieldsBrowserOptions
+                  : undefined
+              }
+              cellActionsOptions={cellActionsOptions}
+              showInspectButton
+              services={{
```

Comment:
Could you please memoize all the non-primitive objects before passing to the components. This will otherwise result in unneeded renders because each time new instance of that object will be created.

Even if the object is `memoized` lower in the tree, it would not help since the identity of the object will anyways change at this component.

Reply:
Sure!